cmd/dagger/module_inspect.go

```diff
@@ -57,24 +59,24 @@ func initializeModule(
 	ctx context.Context,
 	dag *dagger.Client,
 	srcRef string,
-	doFindUp bool,
 	srcOpts ...dagger.ModuleSourceOpts,
 ) (rdef *moduleDef, rerr error) {
 	ctx, span := Tracer().Start(ctx, "load module")
 	defer telemetry.End(span, func() error { return rerr })
 
 	findCtx, findSpan := Tracer().Start(ctx, "finding module configuration", telemetry.Encapsulate())
-	conf, err := getModuleConfigurationForSourceRef(findCtx, dag, srcRef, doFindUp, true, srcOpts...)
+	modSrc := dag.ModuleSource(srcRef, srcOpts...)
+	configExists, err := modSrc.ConfigExists(findCtx)
```

Comment:
Doesn't `ConfigExists` only make sense when you have `AllowNotExists` now? So the `!configExists` below won't ever be hit because if `srcRef` isn't pointing to a module then this will return an error instead.

I think we can rename `initializeModuleConfig` to `initializeModule` and replace the other one with:
```diff
- initializeModule(ctx, dag, modRef, opts)
+ initializeModule(ctx, dag, dag.ModuleSource(modRef, opts))
```

Reply:
I meant to use replace the `modRef` and `opts` arguments with a single `ModuleSource` argument (see first comment).

Then I also meant to keep `tryInitializeModule` as a convenience for adding the `AllowNotExists` option and using `ConfigExists`, but maybe now it's not necessary (see https://github.com/dagger/dagger/pull/9505#discussion_r1959634827).